src/schemas/workflowSchemas.ts

```diff
@@ -123,3 +124,111 @@ export const workflowSchema = z.object({
+    "dark-blue",
+    "light-blue",
+    "sky-blue",
+    "mint-green",
+    "persian-green",
+    "dark-green",
+    "light-green",
+    "yellow",
+    "pink",
+    "orange",
+    "brown",
+  ])
+  .describe("Color of the workflow step displayed in the UI");
+
+// Transition reference schema (step codename reference)
+const transitionToStepSchema = z.object({
+  codename: z.string().optional().describe("Target step codename"),
+  id: z.string().optional().describe("Target step ID (for update operations)"),
+});
+
+const transitionToSchema = z.object({
+  step: transitionToStepSchema.describe("Reference to the target step"),
+});
+
+// Workflow step input schema (for creating/updating workflows)
+export const workflowStepInputSchema = z.object({
+  id: z
+    .string()
+    .uuid()
+    .optional()
+    .describe("Step ID (required when updating existing steps)"),
+  name: z.string().describe("Human-readable name of the workflow step"),
+  codename: z.string().describe("Codename for API operations"),
+  color: workflowStepColorSchema,
+  transitions_to: z
+    .array(transitionToSchema)
+    .describe("Array of step references this step can transition to"),
+  role_ids: z
+    .array(z.string())
+    .describe("Array of role IDs that have permissions for this step"),
+});
+
+// Published step input schema
+export const publishedStepInputSchema = z.object({
+  id: z.string().optional().describe("Published step ID"),
+  name: z.string().optional().describe("Published step name"),
+  codename: z.string().optional().describe("Published step codename"),
+  unpublish_role_ids: z
+    .array(z.string())
+    .optional()
+    .describe("Array of role IDs that can unpublish content"),
+  create_new_version_role_ids: z
+    .array(z.string())
+    .optional()
+    .describe("Array of role IDs that can create new versions"),
+});
+
+// Archived step input schema
+export const archivedStepInputSchema = z.object({
+  id: z.string().optional().describe("Archived step ID"),
+  name: z.string().optional().describe("Archived step name"),
+  codename: z.string().optional().describe("Archived step codename"),
+  role_ids: z
+    .array(z.string())
+    .optional()
+    .describe("Array of role IDs that can restore archived content"),
+});
+
+// Workflow scope input schema
+export const workflowScopeInputSchema = z.object({
+  content_types: z
+    .array(referenceObjectSchema)
+    .describe("Content types this workflow applies to"),
+  collections: z
+    .array(referenceObjectSchema)
+    .optional()
+    .describe("Collections this workflow applies to"),
+});
+
+// Main add/update workflow schema
+export const workflowInputSchema = z.object({
+  name: z.string().describe("Human-readable name of the workflow"),
+  codename: z
+    .string()
+    .optional()
+    .describe("Codename for API operations (auto-generated if omitted)"),
+  scopes: z
+    .array(workflowScopeInputSchema)
+    .describe("Array of scopes defining which content types use this workflow"),
```

Comment:
Scopes are currently content type and collection. You mention content types only here. Also, this might change in the future. Please simplify the description. It might be worth it to mention what empty array means?

Reply:
I adjusted the description to mention collections too and also mention the behavior of setting both to empty. can't say I simplified anything, I mostly copied the description from the api reference